core/block.js

```diff
@@ -1001,6 +1028,13 @@ Blockly.Block.prototype.jsonInit = function(json) {
     var localizedValue = Blockly.utils.replaceMessageReferences(rawValue);
     this.setHelpUrl(localizedValue);
   }
+  if (Array.isArray(json['extensions'])) {
+    var extensionNames = json['extensions'];
+    for (i = 0; i < extensionNames.length; ++i) {      
```

Comment:
It's dangerous to not redeclare 'i' here.  If the 'i' variable is deleted 30 lines up, then this i suddenly becomes global.  That's why it's a good idea to keep var statements near their use.